### PR TITLE
Dchmyga/tdi 41663 secure saxparserfactory config library

### PR DESCRIPTION
**Pull request by chmyga**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Update talendMQRFH2 and talendSAX libraries to be built with maven; update talendMQRFH2 library version in talendMQConnectionUtil library.

### CODE REVIEW

File: main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java [hunk 1]
```diff
@@ -185,15 +187,18 @@ public void parse(java.io.InputStream is, String charset) {
         Reader reader = null;
         try {
             DefaultHandler hd = null;
-            SAXParser saxParser = SAXParserFactory.newInstance().newSAXParser();
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature(javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING, Boolean.TRUE);
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java]:
```suggestion
            spf.setFeature(javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING, true);
```

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/pom.xml [hunk 2]
```diff
@@ -45,12 +69,76 @@
 			<artifactId>talendMQRFH2</artifactId>
 			<version>1.0.1-20190206</version>
 		</dependency>
+		<dependency>
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/pom.xml]: indentation broken

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/pom.xml [hunk 3]
```diff
@@ -45,12 +69,76 @@
...
+				<enabled>false</enabled>
+			</releases>
+			<snapshots>
+				<enabled>true</enabled>
+			</snapshots>
+			<url>https://artifacts-zl.talend.com/nexus/content/repositories/snapshots/</url>
+		</repository>
+		<repository>
+			<id>talend-update</id>
+			<url>https://talend-update.talend.com/nexus/content/repositories/libraries/</url>
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/pom.xml]: IMO, it's better to extract talend nexus url to pom's property, i.e.:
<url>${talend.nexus.url}/nexus/content/repositories/libraries/</url>

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/main/java/org/talend/mq/MQQueueManagerFactory.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.talend.mq;
+
+import java.util.Hashtable;
+
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQQueueManager;
+
+public interface MQQueueManagerFactory {
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/main/java/org/talend/mq/MQQueueManagerFactory.java]: I don't fully get into why did you add 2 interfaces `ActiveMQConnectionFactoryProvider ` and `MQQueueManagerFactory` and never implement them, could you please explain it in javadoc?

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
-	public void testSameConnNAme() throws JMSException {
-		assertTrue(SharedActiveMQConnection.getMQConnection("tcp://localhost:61616", "", "", "conn")==SharedActiveMQConnection.getMQConnection("tcp://localhost:61616", "", "", "conn"));
-	}
-	@Test
-	public void testDiffConnNAme() throws JMSException {
-		assertFalse(SharedActiveMQConnection.getMQConnection("tcp://localhost:61616", "", "", "conn1")==SharedActiveMQConnection.getMQConnection("tcp://localhost:61616", "", "", "conn2"));
-	}
+
+    @Test
+    public void testSameConnNAme() throws JMSException {
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java]: CamelCase pb in test method name

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java [hunk 6]
```diff
@@ -1,22 +1,41 @@
...
+        ActiveMQConnectionFactoryProvider provider = createConnectionFactoryProvider();
+        assertTrue(SharedActiveMQConnection
+                .getMQConnection("tcp://localhost:61616", "", "", "conn", provider) == SharedActiveMQConnection
+                        .getMQConnection("tcp://localhost:61616", "", "", "conn", provider));
+    }
+
+    @Test
+    public void testDiffConnNAme() throws JMSException {
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java]: And here also

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java [hunk 7]
```diff
@@ -1,22 +1,41 @@
...
+        ActiveMQConnectionFactoryProvider provider = createConnectionFactoryProvider();
+        assertFalse(SharedActiveMQConnection
+                .getMQConnection("tcp://localhost:61616", "", "", "conn1", provider) == SharedActiveMQConnection
+                        .getMQConnection("tcp://localhost:61616", "", "", "conn2", provider));
```
Comment (dmytro-sylaiev) [inline main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java]: Don't sure it's good practice:
1) use `assertFalse(o1 == o2)`
2) use some random hardcoded values (at least you can declare them in local variables)
3) what are the real purpose of this test?

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/main/java/org/talend/mq/MQQueueManagerFactory.java (hunk 4 above)
Comment (chmyga) [reply main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/main/java/org/talend/mq/MQQueueManagerFactory.java]: those are functional interfaces. I've added them to easily test the required classes.

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java (hunk 5 above)
Comment (chmyga) [reply main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java]: Names were present before my changes, but I'll fix them. thanks!

File: main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java (hunk 7 above)
Comment (chmyga) [reply main/plugins/org.talend.designer.components.libs/libs_src/talendMQConnectionUtil/src/test/java/org/talend/mq/TestSharedActiveMQConn.java]: please explain your first point. The purpose of the test is to check those two are two different instances, and not the same shared instance. If we have different connection names, a new connection should be created for every name. But we need to return the same connection instance for the same connection names.

File: main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java [hunk 8]
```diff
@@ -140,21 +142,21 @@ public void parse(String fileURL, String charset) {
         Reader reader = null;
         try {
             DefaultHandler hd = null;
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature(javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING, Boolean.TRUE);
+            spf.setFeature(DISALLOW_DOCTYPE_DECL, true);
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java]: This should be set instead in the if (ignoreDTD) statement. If ignoreDTD is false, then we should allow DTDs.

File: main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/simpleparser/SimpleSAXLooper.java [hunk 9]
```diff
@@ -199,59 +205,59 @@ public void parse(java.io.InputStream is, String charset) {
...
             SAXParser saxParser = null;
-            if(!ignoreDTD) { //orginal code
-            	saxParser = SAXParserFactory.newInstance().newSAXParser();
-            } else {
-	            SAXParserFactory spf = SAXParserFactory.newInstance();
-	            spf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
-				saxParser = spf.newSAXParser();
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature(javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING, Boolean.TRUE);
+            spf.setFeature(DISALLOW_DOCTYPE_DECL, true);
```
Comment (coheigea) [inline main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/simpleparser/SimpleSAXLooper.java]: Move into if statement here as above.

File: main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java (hunk 8 above)
Comment (chmyga) [reply main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java]: @coheigea do you mean setting both features - javax.xml.XMLConstants.FEATURE_SECURE_PROCESSING and DISALLOW_DOCTYPE_DECL ? Or only the last one(DISALLOW_DOCTYPE_DECL)? Also, there is a code in a different method(parse(java.io.InputStream is, String charset) - starting on line 185) which I didn't see before - there is no mention of ignoreDTD. Should we add the same if-else block there?

Comment (coheigea) [reply main/plugins/org.talend.designer.components.libs/libs_src/TalendSAX/src/main/java/org/talend/xml/sax/ComplexSAXLooper.java]: Just the last one, we should set secure processing for both. Yes best to make it consistent - you could have a single factory method to create SAXParser.
